apps/daimo-mobile/src/common/nav.ts

```diff
@@ -146,6 +154,45 @@ export function useOnboardingNav<
 
 export type MainNav = ReturnType<typeof useNav>;
 
+export type OnboardingNav = ReturnType<typeof useOnboardingNav>;
+
+// Handles deeplinks during onboarding, if any.
+export function useOnboardingDeepLinkHandler() {
+  const nav = useOnboardingNav();
+
+  // If we opened the app by clicking an invite link, go straight into onboarding.
+  const navToInitialInviteDeepLinkIfPresent = async () => {
+    const deeplink = await getInitialDeepLink();
+    if (deeplink == null) return;
+    // Use invite / payment link now > don't nav there after onboarding
+    await handleOnboardingDeepLink(nav, deeplink);
+    markInitialDeepLinkHandled();
+  };
+
+  useEffect(() => {
+    navToInitialInviteDeepLinkIfPresent();
+    const listener: URLListener = ({ url }) =>
+      handleOnboardingDeepLink(nav, url);
+    const sub = addEventListener("url", listener);
+    return () => sub.remove();
```

Comment:
just to double check this doesn’t break deep links if you switch tabs to another one right?